### PR TITLE
Workbench application and space navigation

### PR DESCRIPTION
**Pull request by z2sx**

### CODE REVIEW

File: packages/core/src/memdb.ts [hunk 1]
```diff
@@ -66,9 +66,13 @@ class MemDb {
 
   async findAll<T extends Doc>(_class: Ref<Class<T>>, query: DocumentQuery<T>): Promise<T[]> {
     let result: Doc[]
-    if (query._id !== undefined) {
-      const obj = this.objectById.get(query._id)
-      result = obj !== undefined ? [obj] : []
+    if (Object.prototype.hasOwnProperty.call(query, '_id')) {
```
Comment (haiodo) [inline packages/core/src/memdb.ts:69]: Why query._id not work?

Comment (z2sx) [reply packages/core/src/memdb.ts:69]: do not work for queries like { _id: undefined }

Comment (z2sx) [reply packages/core/src/memdb.ts:69]: consider I have a query `{ _id: currentApp}`, so when `currentApp === undefined`, I think this query should return nothing. It was returning all the apps without this fix.

File: plugins/workbench-impl/src/components/navigator/SpacesNav.svelte [hunk 2]
```diff
@@ -46,12 +47,19 @@
       showModal(model.createComponent, {})
     }
   }
+
+  function selectSpace(id: Ref<Space>) {
+    const loc = getCurrentLocation()
+    loc.path[2] = id
```
Comment (haiodo) [inline plugins/workbench-impl/src/components/navigator/SpacesNav.svelte:53]: I'm personally not like this approach, it is not flexible, how it work if not enought arguments are provided?

Comment (z2sx) [reply plugins/workbench-impl/src/components/navigator/SpacesNav.svelte:53]: I do not like this too, please suggest anything to make it good. Also I do not understand what you mean by "not enought arguments are provided"?

Comment (haiodo) [reply plugins/workbench-impl/src/components/navigator/SpacesNav.svelte:53]: How we could be sure we have at least 2 values in loc.path?

Comment (haiodo) [reply plugins/workbench-impl/src/components/navigator/SpacesNav.svelte:53]: Routes stuff you removed was my proposal :-)

Comment (z2sx) [reply plugins/workbench-impl/src/components/navigator/SpacesNav.svelte:53]: Let's discuss this on meeting, I tried to use routes stuff during weekend but it does not work well for me

File: packages/core/src/memdb.ts (hunk 1 above)
Comment (haiodo) [reply packages/core/src/memdb.ts:69]: Ok.

**Pull request merged**